packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -57,6 +56,154 @@ class NetworkController extends DisposableController
     subscribeToFilterChanges();
   }
 
+  String? exportAsHarFile() {
+    httpRequests =
+        filteredData.value.whereType<DartIOHttpRequestData>().toList();
+
+    if (httpRequests!.isEmpty) {
+      debugPrint('No valid request data to export');
+      return '';
+    }
+
+    try {
+      if (httpRequests!.isNotEmpty) {
+        final har = {
```

Comment:
Nit: these maps and lists should have types associated with them to avoid introducing `dynamic`.

Reply:
httpRequests is of type 'List<DartIOHttpRequestData>?'
which maps and list did you mean?